lib/transmuxer/ts_transmuxer.js

```diff
@@ -685,6 +690,93 @@ shaka.transmuxer.TsTransmuxer = class {
       stream: stream,
     };
   }
+
+
+  /**
+   * @param {shaka.util.TsParser} tsParser
+   * @param {shaka.extern.Stream} stream
+   * @param {number} duration
+   * @return {shaka.util.Mp4Generator.StreamInfo}
+   * @private
+   */
+  getHvcStreamInfo_(tsParser, stream, duration) {
+    const H265 = shaka.transmuxer.H265;
+    const timescale = shaka.util.TsParser.Timescale;
+
+    /** @type {!Array.<shaka.util.Mp4Generator.Mp4Sample>} */
+    const samples = [];
+
+    /** @type {?number} */
+    let baseMediaDecodeTime = null;
+
+    let nalus = [];
+    const videoData = tsParser.getVideoData();
+    for (let i = 0; i < videoData.length; i++) {
+      const pes = videoData[i];
+      let nextPes;
+      if (i + 1 < videoData.length) {
+        nextPes = videoData[i + 1];
+      }
+      const dataNalus = tsParser.parseNalus(pes, nextPes);
+      nalus = nalus.concat(dataNalus);
+      const frame = H265.parseFrame(dataNalus);
+      if (!frame) {
+        continue;
+      }
+      if (baseMediaDecodeTime == null && pes.dts != null) {
+        baseMediaDecodeTime = pes.dts;
+      }
+      let duration;
+      if (i + 1 < videoData.length) {
+        duration = (videoData[i + 1].dts || 0) - (pes.dts || 0);
+      } else {
+        duration = (pes.dts || 0) - (videoData[i - 1].dts || 0);
+      }
+      samples.push({
+        data: frame.data,
+        size: frame.data.byteLength,
+        duration: duration,
+        cts: Math.round((pes.pts || 0) - (pes.dts || 0)),
```

Comment:
Is this the same as a "composition time offset" (CTO)?

Reply:
Yes! But changing the name implies changing the name in many parts of the application, if you want we can change it but in another PR.

Reply:
Not, I'm okay with it for now.  And it's internal, so we can always change it later without breaking compatibility, right?

Reply:
Yes it’s internal